### PR TITLE
Added bookings

### PR DESCRIPTION
**Pull request by tejasraibagi26**
User can now book venues.
Cost is calculated based on the total hours of play.

You can ignore the comments and reviews file they are just re-uploaded as prettier changed single quotes to double.

### CODE REVIEW

File: public/js/err.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const errDiv = document.getElementById("err");
+
+errDiv.addEventListener("click", function () {
```
Comment (troytomasch) [inline public/js/err.js:3]: This is giving an error client side on feed page

File: public/js/register.js [hunk 2]
```diff
@@ -144,7 +124,6 @@ if (registerForm) {
     } catch (e) {
       error.innerHTML = e;
       error.hidden = false;
-      event.preventDefault();
```
Comment (troytomasch) [inline public/js/register.js:147]: Can't remove this as it then allows the route to continue in the case of an error.

File: public/js/bookings.js [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+});
+
+endTime.addEventListener("change", (e) => {
+  eT = e.target.value.split(":")[0];
+  updateCost();
+});
+
+function updateCost() {
+  total = (eT - sT) * price.value;
+  console.log(total);
```
Comment (troytomasch) [inline public/js/bookings.js:23]: Missed this

**Review by troytomasch — CHANGES_REQUESTED** [review]

**Comment by troytomasch** [comment]
Also, need to merge main to resolve conflicts

**Comment by tejasraibagi26** [comment]
> Also, need to merge main to resolve conflicts

Good to go now!

File: public/js/bookings.js (hunk 3 above)
Comment (tejasraibagi26) [reply public/js/bookings.js:23]: This is basically to calculate the total cost on the fly. You might change the time period. eT here is end time and sT is start time. I'll add some documentation.

**Comment by troytomasch** [comment]
I'm all good to merge it once you fix the couple comments I had.

File: public/js/register.js (hunk 2 above)
Comment (tejasraibagi26) [reply public/js/register.js:147]: okay! Will add this!

File: public/js/err.js (hunk 1 above)
Comment (tejasraibagi26) [reply public/js/err.js:3]: What error?

File: public/js/bookings.js (hunk 3 above)
Comment (troytomasch) [reply public/js/bookings.js:23]: No worries. I was just referring to the console.log(total) haha.

File: public/js/err.js (hunk 1 above)
Comment (troytomasch) [reply public/js/err.js:3]: image

Comment (tejasraibagi26) [reply public/js/err.js]: Fixed it.

File: public/js/register.js [hunk 4]
```diff
@@ -33,6 +12,7 @@ if (registerForm) {
   const error = document.getElementById("registerError");
 
   registerForm.addEventListener("submit", (event) => {
+    e.preventDefault();
```
Comment (troytomasch) [inline public/js/register.js:15]: This can be removed right?

File: views/user/register.handlebars [hunk 5]
```diff
@@ -3,7 +3,13 @@
     <h1 id="registerTitle">Register</h1>
     <h3 id="registerError" hidden></h3>
     <h3 id="registerError"> {{error}}</h3>
-    <form action="/user/create" name="create" method="post" id="registerForm">
+    <form
+      action="/user/create"
+      name="create"
+      onsubmit="return validate()"
```
Comment (troytomasch) [inline views/user/register.handlebars:9]: The onsubmit function here throws an error

File: public/js/register.js (hunk 4 above)
Comment (tejasraibagi26) [reply public/js/register.js:15]: Yup. Also i removed on submit from registration form. As we don't use that.

File: views/user/register.handlebars (hunk 5 above)
Comment (tejasraibagi26) [reply views/user/register.handlebars]: Fixed it

**Review by troytomasch — APPROVED** [review]

**Pull request merged**